types/preview/@ember/owner/index.d.ts

```diff
@@ -18,6 +17,7 @@ declare module '@ember/owner' {
     /**
      * Given a {@linkcode FullName} return a corresponding instance.
      */
+    lookup<Name extends keyof Registry>(fullName: `service:${Name}`): Registry[Name];
```

Comment:
Given the overall change in direction, this will update to be like this:

```suggestion
    lookup<Name extends keyof Registry>(fullName: `service:${Name}`): InstanceType<Registry[Name]>;
```

We can also do the same (with a little work) for `factoryFor` and its `FactoryManager` return type: the `factoryFor` will end up able to return the right thing as well.